core.c

```diff
@@ -336,29 +330,28 @@ void core_reset() {
 /**
  * @name	core_check_gl_error
  * @brief	check's for a gl error and reports it to the platform
- * @retval	NONE
+ * @retval	(bool) whether or not there was an error
  */
-void core_check_gl_error() {
+bool core_check_gl_error() {
```

Comment:
The function name here is really misleading. It says, in english, check if there was a GL error and tell me yes or no. However, there are a bunch of side effects in here. Maybe an in-out param can be added which is set to the error code if there was an error or -1 otherwise. The caller is then responsible for checking if it was a memory warning or whatever else.

```c
int gl_error;
if (core_check_gl_error(&gl_error)) {
  switch gl_error:
    case GLERROR_OUT_OF_MEMORY:
      // ...
      break;
    // ...
}
```